sdk/core/System.ClientModel/src/Internal/Logging/LoggingHandler.cs

```diff
@@ -0,0 +1,272 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Text;
+using Microsoft.Extensions.Logging;
+
+namespace System.ClientModel.Internal;
+
+internal class LoggingHandler
+{
+    private ILogger _logger;
+    private PipelineMessageSanitizer _sanitizer;
+    private ClientModelLogMessages _logMessages;
+
+    public LoggingHandler(ILogger logger, PipelineMessageSanitizer sanitizer)
+    {
+        _logger = logger;
+        _sanitizer = sanitizer;
+        _logMessages = new ClientModelLogMessages(logger);
+    }
+
+    public void LogRequest(PipelineRequest request, string requestId, string? assemblyName)
+    {
+        bool isEventSourceEnabled = ClientModelEventSource.Log.IsEnabled(EventLevel.Informational, EventKeywords.None);
+        bool isLoggerEnabled = _logger.IsEnabled(LogLevel.Information);
+
+        if (isLoggerEnabled)
```

Comment:
Probably should change this to if ILogger is not null, and then check if its enabled. We shouldn't be inconsistent. i.e. someone has an event listener (enabled for verbose) and also passes an ILogger (enabled for warnings). Then they will see warnings to ILogger and everything BUT warnings to event source. 
Should probably just have one bool that is set in the constructor or something.